install_config/storage_examples/containerized_heketi_with_dedicated_gluster.adoc

```diff
@@ -0,0 +1,379 @@
+
+[[containerized-heketi-dedicated-gluster-example-prerequisites]]
+== Environment and Prerequisites
+This example uses the following environment and prerequisites:
+
+* GlusterFS cluster running Red Hat Gluster Storage (RHGS) 3.1. Three nodes, each with at least two 100GB RAW devices:
+** *gluster23.rhs* (192.168.1.200)
+** *gluster24.rhs* (192.168.1.201)
+** *gluster25.rhs* (192.168.1.202)
+* This example uses an all-in-one {product-title} cluster (master and node on a single host), though it can work using a standard, multi-node cluster as well.
+** *k8dev2.rhs* (192.168.1.208)
+
+[[containerized-heketi-dedicated-gluster-example-install-heketi]]
+== Installing and Configuring Heketi
+Heketi is used to manage the Gluster cluster storage (adding volumes, removing volumes, etc.). Download link:https://github.com/heketi/heketi/blob/master/extras/openshift/templates/deploy-heketi-template.json[`deploy-heketi-template`] to install Heketi on OpenShift.
+
+[NOTE]
+====
+This template file places the database in an EmptyDir volume. Adjust the database accordingly for a reliable persistent storage.
+====
+
+. Create a new project:
++
+[source, bash]
+----
+$ oc new-project <project-name>
+----
+
+. Enable privileged containers in the new project:
++
+[source, bash]
+----
+$ oc adm policy add-scc-to-user privileged -z default
+----
+
+. Register the `deploy-heketi` template:
++
+[source, bash]
+----
+$ oc create -f <template-path>/deploy-heketi-template
+----
+
+. Deploy the bootstrap Heketi container:
++
+[source, bash]
+----
+$ oc process deploy-heketi -v \
+         HEKETI_KUBE_NAMESPACE=<project-name> \
+         HEKETI_KUBE_APIHOST=<master-url-and-port> \
+         HEKETI_KUBE_INSECURE=y \
+         HEKETI_KUBE_USER=<cluster-admin-username> \
+         HEKETI_KUBE_PASSWORD=<cluster-admin-password> | oc create -f -
+----
+
+. Wait until the `deploy-heketi` pod starts and all services are running. Then get Heketi service details:
++
+[source, bash]
+----
+$ oc get svc
+NAME            CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+deploy-heketi   172.30.96.173   <none>        8080/TCP   2m
+----
+
+. Check if Heketi services are running properly, it must return `Hello from Heketi`.
++
+[source, bash]
+----
+$ curl http://<cluster-ip>:8080/hello
+Hello from Heketi
+----
+
+. Set an environment variable for the Heketi server:
++
+[source, bash]
+----
+$ export HEKETI_CLI_SERVER=http://<cluster-ip>:8080
+----
+
+[[dedicated-glusterfs-dynamic-example-loading-topology]]
+== Loading Topology
+
+Topology is used to tell Heketi about the environment and what nodes and devices
+it will manage.
+
+[NOTE]
+====
+Heketi is currently limited to managing raw devices only. If a device is already
+a Gluster volume, it is skipped and ignored.
+====
+
+. Create and load the topology file. There is a sample file located in
+*_/usr/share/heketi/toplogy-sample.json_* by default, or *_/etc/heketi_*
```

Comment:
Didn't see /usr/share/heketi/toplogy-sample.json, may because I just have heketi-client RPM package installed. Should hint customer here.

Reply:
Thanks @wkshi I have updated the docs to include more details around this one by adding following note:
```
Depending upon your method of installation this file may not exist. If its missing, manually create the *_toplogy-sample.json_* file, as shown in the following example.
```

Reply:
Looks good to me~